### ISSUE TITLE
ERC20 transferFrom

### ISSUE THREAD

**Issue by sanderkruger** [Adamant-im/ETH-transactions-storage#4]
Looking at the code and the database model, it seems that the indexer doesn't support indexing ERC20 token transactions if the transferFrom function is used. Could this support be added?

**Comment by adamant-al** [comment]
Hi,

currently, we identify ERC20 transactions as:
`if inputinfo.startswith('0xa9059cbb')`

Doesn't `transferFrom` function write `0xa9059cbb` into `inputinfo`?
If no, how `transferFrom` transactions can be identified and parsed?

**Comment by sanderkruger** [comment]
Hi,

I haven't tested or tried it out, but in an online keccak-256 hasher, I found that the first 8 bytes of the hash of `transferFrom(address,address,uint256)` is `0x23b872dd`, so I think that's what identifies a `transferFrom` transaction.

The other reason I submitted this comment is that I think the issue is that there is no `contract_from` field in the database. For `transfer` transactions, the sender of the tokens is the same as the sender of the transactions, but for transferFrom this is not the case. If transferFrom is used to move ERC20 tokens, then the sender address is passed as a parameter, then the recipient and finally the value. So I guess, contract_from would be `inputinfo[10:-64]`, contract_to `inputinfo[74:-64]` and contract_value `inputinfo[138:]`.

Hope this helps.
